jaeger-core/src/test/java/io/jaegertracing/internal/JaegerSubclassTest.java

```diff
@@ -0,0 +1,179 @@
+          new HashMap<>(tracer.tags()),
+          tracer.isZipkinSharedRpcSpan(),
+          tracer.scopeManager(),
+          tracer.getBaggageRestrictionManager(),
+          tracer.isExpandExceptionLogs());
+    }
+
+    public Tracer(
+        String serviceName,
+        Reporter reporter,
+        Sampler sampler,
+        PropagationRegistry registry,
+        Clock clock,
+        Metrics metrics,
+        Map<String, Object> tags,
+        boolean zipkinSharedRpcSpan,
+        ScopeManager scopeManager,
+        BaggageRestrictionManager baggageRestrictionManager,
+        boolean expandExceptionLogs) {
+      super(
+          serviceName,
+          reporter,
+          sampler,
+          registry,
+          clock,
+          metrics,
+          tags,
+          zipkinSharedRpcSpan,
+          scopeManager,
+          baggageRestrictionManager,
+          expandExceptionLogs);
+    }
+
+    @Override
+    public SpanBuilder buildSpan(String operationName) {
+      return new SpanBuilder(operationName);
+    }
+  }
+
+  private static class Span extends JaegerSpan {
+    private final SpanContext ctx;
+
+    Span(JaegerSpan span) {
+      this(
+          span.getTracer(),
+          span.getOperationName(),
+          new SpanContext(span.context()),
+          span.getStart(),
+          span.getStartTimeNanoTicks(),
+          span.getComputeDurationViaNanoTicks(),
+          span.getTags(),
+          span.getReferences());
+    }
+
+    public Span(
+        JaegerTracer tracer,
+        String operationName,
+        JaegerSpanContext context,
+        long startTimeMicroseconds,
+        long startTimeNanoTicks,
+        boolean computeDurationViaNanoTicks,
+        Map<String, Object> tags,
+        List<Reference> references) {
+      super(
+          tracer,
+          operationName,
+          context,
+          startTimeMicroseconds,
+          startTimeNanoTicks,
+          computeDurationViaNanoTicks,
+          tags,
+          references);
+      ctx = new SpanContext(context);
+    }
+
+    @Override
+    public SpanContext context() {
+      return ctx;
+    }
+  }
+
+  private static class SpanContext extends JaegerSpanContext {
+    SpanContext(JaegerSpanContext ctx) {
+      this(
+          ctx.getTraceId(),
+          ctx.getSpanId(),
+          ctx.getParentId(),
+          ctx.getFlags(),
+          ctx.baggage(),
+          ctx.getDebugId());
+    }
+
+    public SpanContext(
+        long traceId,
+        long spanId,
+        long parentId,
+        byte flags,
+        Map<String, String> baggage,
+        String debugId) {
+      super(traceId, spanId, parentId, flags, baggage, debugId);
+    }
+  }
+
+  @Test
+  public void testTracer() {
+    final Configuration config = new Configuration("test-service");
+    final Tracer tracer = config.getTracer();
+    final Scope scope = tracer.buildSpan("test-operation").startActive(true);
+    Assert.assertNotNull(tracer.scopeManager().active());
```

Comment:
I understand that if this assertion works, then a bunch of other stuff has also worked, but it would be nice to have your "base" assertions made explicit here.

Reply:
Ya this test isn't perfect. Needs work.

Reply:
Should be a bit better now.